test/y2storage/boot_requirements_errors_test.rb

```diff
@@ -404,66 +318,86 @@
       let(:efiboot) { false }
       let(:prep_id) { Y2Storage::PartitionId::PREP }
 
-      before do
-        allow(storage_arch).to receive(:ppc_power_nv?).and_return(power_nv)
-      end
-
       let(:power_nv) { false }
 
       context "when there is no root" do
-        let(:root_filesystem) { nil }
+        let(:scenario) { "false-swaps" }
         include_examples "unknown boot disk"
       end
 
       context "in a non-PowerNV system (KVM/LPAR)" do
         let(:power_nv) { false }
 
         context "with a partitions-based proposal" do
-          let(:use_lvm) { false }
-          include_examples "PReP partition"
+
+          context "there is a PReP partition" do
+            let(:scenario) { "prep" }
+            include_examples "no errors"
+          end
+
+          context "PReP partition missing" do
+            let(:scenario) { "trivial" }
+            include_examples "missing prep partition"
+          end
         end
 
         context "with a LVM-based proposal" do
-          let(:use_lvm) { true }
-          include_examples "PReP partition"
+          context "there is a PReP partition" do
+            let(:scenario) { "prep_lvm" }
+            include_examples "no errors"
+          end
+
+          context "PReP partition missing" do
+            let(:scenario) { "trivial_lvm" }
+            include_examples "missing prep partition"
+          end
         end
 
-        context "with a Software RAID proposal" do
-          let(:use_raid) { true }
-          include_examples "PReP partition"
+        # sorry, but I won't write it in xml and yaml does not support it
+        # scenario generator would be great
+        xcontext "with a Software RAID proposal" do
```

Comment:
I guess you could use whatever devicegraph we already have with a MD RAID and manually create a PReP partition in before section.

Reply:
this is too artificial for me. My goal is to have clear scenario and then apply test on top of it. Having hacks that modify partition in code makes it just harder to see point of test and verify it is correct.